### PR TITLE
Instructions for DCO

### LINKED ISSUE #362: CLA should be triggered if any contribution from community
Currently, we have a contribution from community, #361.
But his pr didn't trigger our CI, we have to approve it manually.
Not sure CLA will fix it.

### PR DESCRIPTION
**Pull request by ghzlatarev**
## Description

closes: #362

* If DCO check fails the instructions are commented in the PR , not just in the CI output:

Please signoff on all commits with your name, email and gpg key for the Developer's Certificate of Origin.
    `git config --global user.name your_name`
    `git config --global user.email you_email`
COMMAND : `git commit -s -S -m your_commit_message`
    `-s` = `Signed-off-by`
    `-S` = `Verify commit using gpg key`
If there is more than one commit in your pull request and your git client is modern enough (2.13+), rebase the required number of commits with --signoff:
   `git rebase --signoff HEAD~<number_of_commits>` 
Then, force push:
   `git push -f origin <your_branch>`
For instructions on managing gpg signature verification please visit: https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits

---

* Check the comments in the PR for actual example

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (`manta` or `dolphin`) with right title (start with [Manta] or [Dolphin]),
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests.
- [x] Updated relevant documentation in the code.
- [x] Added **one** line describing your change in `<branch>/CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [x] If runtime changes, need to update the version numbers properly:
   * `authoring_version`: The version of the authorship interface. An authoring node will not attempt to author blocks unless this is equal to its native runtime.
   * `spec_version`: The version of the runtime specification. A full node will not attempt to use its native runtime in substitute for the on-chain Wasm runtime unless all of spec_name, spec_version, and authoring_version are the same between Wasm and native.
   * `impl_version`: The version of the implementation of the specification. Nodes are free to ignore this; it serves only as an indication that the code is different; as long as the other two versions are the same then while the actual code may be different, it is nonetheless required to do the same thing. Non-consensus-breaking optimizations are about the only changes that could be made which would result in only the impl_version changing.
   * `transaction_version`: The version of the extrinsics interface. This number must be updated in the following circumstances: extrinsic parameters (number, order, or types) have been changed; extrinsics or pallets have been removed; or the pallet order in the construct_runtime! macro or extrinsic order in a pallet has been changed. If this number is updated, then the spec_version must also be updated 
- [x] If needed, notify the committer this is a draft-release and a tag is needed after merging the PR.
- [x] Verify benchmarks & weights have been updated for any modified runtime logics
- [x] If needed, bump `version` for every crate.
- [x] If import a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- [x] If needed, update our Javascript/Typescript APIs. These APIs are offcially used by exchanges or community developers.
- [x] If we're going to issue a new release, freeze the code one week early(it depends, but usually it's one week), ensure we have enough time for related testing.
- [x] Check if inheriting any upstream runtime storage migrations. If any, perform tests with `try-runtime`.

### CODE REVIEW

**Comment by stechu** [comment]
Can you put this in a contribution guideline markdown?
something like:
https://github.com/atom/atom/blob/master/CONTRIBUTING.md

**Comment by ghzlatarev** [comment]
> Can you put this in a contribution guideline markdown? something like: https://github.com/atom/atom/blob/master/CONTRIBUTING.md

Will do.

**Review by Dengjianping — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+> **Note:** If you find a **Closed** issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
+
+#### How Do I Submit A (Good) Bug Report?
+
+Bugs are tracked as [GitHub issues](https://guides.github.com/features/issues/). 
+
+Explain the problem and include additional details to help maintainers reproduce the problem:
+
+* **Use a clear and descriptive title** for the issue to identify the problem.
+* **Describe the exact network configuraiton** include parachain node version, relay chain node version and any tooling you may be using, such as polkadot-launch and its configurations.
```
Comment (Garandor) [inline CONTRIBUTING.md]: typo configuraiton

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Furthermore, provide more context by answering these questions:
+
+* **Did the problem start happening recently** (e.g. after updating to a new version of Manta/Calamari) or was this always a problem?
+* If the problem started happening recently, **can you reproduce the problem in an older version of Manta/Calamari?** What's the most recent version in which the problem doesn't happen? You can download older versions from [the releases page](https://github.com/Manta-Network/Manta/releases).
+* **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+### Always include details about your configuration and environment
+
+* **Are you working with the Manta/Calamari mainnets or have you depoloyed locally**
+* **Wich version of Manta/Calamari are you using?**
```
Comment (Garandor) [inline CONTRIBUTING.md]: typo, Wich

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+Furthermore, provide more context by answering these questions:
+
+* **Did the problem start happening recently** (e.g. after updating to a new version of Manta/Calamari) or was this always a problem?
+* If the problem started happening recently, **can you reproduce the problem in an older version of Manta/Calamari?** What's the most recent version in which the problem doesn't happen? You can download older versions from [the releases page](https://github.com/Manta-Network/Manta/releases).
+* **Can you reliably reproduce the issue?** If not, provide details about how often the problem happens and under which conditions it normally happens.
+
+### Always include details about your configuration and environment
+
+* **Are you working with the Manta/Calamari mainnets or have you depoloyed locally**
```
Comment (Garandor) [inline CONTRIBUTING.md]: depoloyed

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+### Suggesting Enhancements
+
+This section guides you through submitting an enhancement suggestion for Manta/Calamari, including completely new features and minor improvements to existing functionality. Following these guidelines helps maintainers and the community understand your suggestion :pencil: and find related suggestions :mag_right:.
+
+Before creating enhancement suggestions, please check the existing issues as you might find out that you don't need to create one. When you are creating an enhancement suggestion, please [include as many details as possible](#how-do-i-submit-a-good-enhancement-suggestion).  including the steps that you imagine you would take if the feature you're requesting existed.
+
+#### How Do I Submit A (Good) Enhancement Suggestion?
+
+Enhancement suggestions are tracked as [GitHub issues](https://guides.github.com/features/issues/).Create an issue on that repository and provide the following information:
```
Comment (Garandor) [inline CONTRIBUTING.md]: space after dot

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+1. Every commit needs to be signed-off with your name, email and [gpg key](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits):
+    1.1. `git config --global user.name your_name`
+    1.2. `git config --global user.email you_email`
+    1.3. `git commit -s -S -m your_commit_message`
+        * `-s` = `Signed-off-by`
+        * `-S` = `Verify commit using gpg key`
+2. Follow all instructions in [the template](.github/PULL_REQUEST_TEMPLATE.md)
+3. After you submit your pull request, verify that all CI checks are passing <details><summary>What if CI checks are failing?</summary>If a CI check is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. A maintainer will re-run the status check for you. If we conclude that the failure was a false positive, then we will open an issue to track that problem.</details>
+4. If you have a significant design decision to make please document it as an ehancement request first, so we can review it and provide feedback before any code is written.
```
Comment (Garandor) [inline CONTRIBUTING.md]: ehancement

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+While the prerequisites above must be satisfied prior to having your pull request reviewed, the reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted.
```
Comment (Garandor) [inline CONTRIBUTING.md]: isnt "must be satisfied" a bit too strong for a "guideline"?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+
+## Styleguides
+
+* We generally follow the [Rust styleguide](https://doc.rust-lang.org/1.0.0/style/).
+
+### Comments
+
+* Comments should focus more on the `why` instead of the `what` is the code.
+* Write comments on top of your code.
```
Comment (Garandor) [inline CONTRIBUTING.md:96]: we don't do inline comments to the right of a code line? Good to know :D

**Review by Garandor — CHANGES_REQUESTED** [review]
Pretty clear instructions, just found a few typos to fix

**Review by grenade — APPROVED** [review]

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+The following is a set of guidelines for contributing to the Manta Network codebase. These are mostly guidelines, not rules. Use your best judgment, and feel free to propose changes to this document in a pull request.
+
+## How Can I Contribute?
+
+### Reporting Bugs
+
+Following these guidelines helps maintainers and the community understand your report :pencil:, reproduce the behavior :computer: :computer:, and find related reports :mag_right:.
+
+* **Perform a cursory search in the Manta Network organization to see if the problem has already been reported. If it has **and the issue is still open**, add a comment to the existing issue instead of opening a new one.
```
Comment (Garandor) [inline CONTRIBUTING.md]: this has one too many sets of **

**Review by Garandor — APPROVED** [review]

**Pull request merged**